components/CreateReplyRequestDialog/CreateReplyRequestDialog.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
+
+const localStorage = typeof window === 'undefined' ? {} : window.localStorage;
+const formInitialState = {
+  visible: false,
+  disabled: true,
+  text: '',
+};
+
+const CREATE_REPLY_REQUEST = gql`
+  mutation CreateReplyRequestFromForm($articleId: String!, $reason: String!) {
+    CreateReplyRequest(articleId: $articleId, reason: $reason) {
+      id
+    }
+  }
+`;
+
+function SubmitButton({ articleId, text, disabled }) {
+  const [createReplyRequest] = useMutation(CREATE_REPLY_REQUEST, {
+    refetchQueries: ['LoadArticlePage'],
+  });
+
+  const handleSubmit = e => {
+    e.preventDefault(); // prevent reload
+    if (disabled) return;
+    createReplyRequest({ variables: { articleId, reason: text } });
+  };
+
+  return (
+    <button onClick={handleSubmit} disabled={disabled}>
+      送出
+    </button>
+  );
+}
+
+class CreateReplyRequestDialog extends React.PureComponent {
+  static defaultProps = {};
+
+  constructor() {
+    super();
+    this.state = {
+      ...formInitialState,
+    };
+  }
+
+  componentDidMount() {
+    const { text } = this.state;
+
+    // restore from localStorage if applicable.
+    // We don't do this in constructor to avoid server/client render mismatch.
+    //
+    this.setState({
+      text: localStorage.text || text,
+    });
+  }
+
+  set(key, value) {
+    this.setState({ [key]: value });
+
+    // Backup to localStorage
+    requestAnimationFrame(() => (localStorage[key] = value));
+  }
+
+  handleTextChange = ({ target: { value } }) => {
+    this.setState({
+      text: value,
+      disabled: !value || value.length < 80,
+    });
+  };
+
+  showForm = () => {
+    this.setState({ visible: true });
+  };
+
+  render = () => {
+    const { text, visible, disabled } = this.state;
+
+    return (
+      <div>
```

Comment:
I think it looks OK, but currently users cannot hide the "dialog" after opening it.

I suggest either we add a cancel button, or we directly use a [simple dialog](https://material-ui.com/components/dialogs/#simple-dialogs) and provide its `onClose` callback.